src/Sentry/Android/JavaExtensions.cs

```diff
@@ -0,0 +1,163 @@
+        }
+
+        public static Java.SentryLevel ToJavaSentryLevel(this SentryLevel level) =>
+            level switch
+            {
+                SentryLevel.Debug => Java.SentryLevel.Debug!,
+                SentryLevel.Info => Java.SentryLevel.Info!,
+                SentryLevel.Warning => Java.SentryLevel.Warning!,
+                SentryLevel.Error => Java.SentryLevel.Error!,
+                SentryLevel.Fatal => Java.SentryLevel.Fatal!,
+                _ => throw new ArgumentException(null, nameof(level))
+            };
+
+        public static SentryId ToSentryId(this Java.Protocol.SentryId sentryId) =>
+            new(Guid.Parse(sentryId.ToString()));
+
+        public static SpanId ToSpanId(this Java.SpanId spanId) =>
+            new(spanId.ToString());
+
+        public static SpanStatus ToSpanStatus(this Java.SpanStatus status)
+        {
+            // note: switch doesn't work here because JNI enums are not constants
+            if (status == Java.SpanStatus.Ok)
+                return SpanStatus.Ok;
+            if (status == Java.SpanStatus.DeadlineExceeded)
+                return SpanStatus.DeadlineExceeded;
+            if (status == Java.SpanStatus.Unauthenticated)
+                return SpanStatus.Unauthenticated;
+            if (status == Java.SpanStatus.PermissionDenied)
+                return SpanStatus.PermissionDenied;
+            if (status == Java.SpanStatus.NotFound)
+                return SpanStatus.NotFound;
+            if (status == Java.SpanStatus.ResourceExhausted)
+                return SpanStatus.ResourceExhausted;
+            if (status == Java.SpanStatus.InvalidArgument)
+                return SpanStatus.InvalidArgument;
+            if (status == Java.SpanStatus.Unimplemented)
+                return SpanStatus.Unimplemented;
+            if (status == Java.SpanStatus.Unavailable)
+                return SpanStatus.Unavailable;
+            if (status == Java.SpanStatus.InternalError)
+                return SpanStatus.InternalError;
+            if (status == Java.SpanStatus.UnknownError)
+                return SpanStatus.UnknownError;
+            if (status == Java.SpanStatus.Cancelled)
+                return SpanStatus.Cancelled;
+            if (status == Java.SpanStatus.AlreadyExists)
+                return SpanStatus.AlreadyExists;
+            if (status == Java.SpanStatus.FailedPrecondition)
+                return SpanStatus.FailedPrecondition;
+            if (status == Java.SpanStatus.Aborted)
+                return SpanStatus.Aborted;
+            if (status == Java.SpanStatus.OutOfRange)
+                return SpanStatus.OutOfRange;
+            if (status == Java.SpanStatus.DataLoss)
+                return SpanStatus.DataLoss;
+
+            throw new ArgumentException(null, nameof(status));
+        }
+
+        public static SentryEvent ToSentryEvent(this Java.SentryEvent sentryEvent, Java.SentryOptions javaOptions)
```

Comment:
why the need to round trip via serialization? can the props of Java.SentryEvent be duplicated into a  SentryEvent  without the serialization?

Reply:
Great question.  So, I started down the path of mapping individual properties, similar to the `ToSentryLevel`/`ToJavaSentryLevel` methods.  I quickly found lots of problems:

- The mapping code was _ginormous_, due to the number of different objects that are part of the object graph. All of these objects would need to be mapped:  `SentryEvent`, `SentryMessage`, `SentryValues`, `SentryException`, `SentryStackTrace`, `SentryStackFrame`, `Mechanism`, `SentryThread`, `DebugImage`, `Contexts`, `User`, `SdkVersion`, `BreadCrumb`, `App`, `Browser`, `Device`, `DeviceOrientation`, `OperatingSystem`, `Runtime`, `Gpu`, `Trace`, `SpanId`, `SentryId`, `SpanStatus` (perhaps more).
- There are lots of places with collections of `object` typed properties.  They could end up containing usable primitives, but they could also contain references to Java objects and might be difficult to work with in .NET - especially since we have made all of Sentry's Java public API *internal* for .NET in the JNI transforms.
- The objects are not completely the same on both sides.  Trying to handle all the differences leads to fragile code that likely will miss updates.  Through serialization, we at least get a contract that both sides use the same way, as it's the same that is required for shipping data to Sentry.

I recognize there may be some performance disadvantages to this approach, so we may want to map *some* of the object graph, or perhaps choose to use the original object for the return value if its clone hasn't been modified (if we can detect that).  But for now, this works.

Reply:
ok. thanks for the clarification

Reply:
I actually had an idea that might avoid the serialization cost. I'll try it out before going into details just yet.  Stay tuned...

Reply:
I'll come back to the serialization issue in a separate PR.  This works for now, and doesn't even matter if you aren't using `BeforeSend` or `BeforeBreadcrumb` delegates.

Reply:
I'll come back to the serialization issue in a separate PR.  This works for now, and doesn't even matter if you aren't using `BeforeSend` or `BeforeBreadcrumb` delegates.